pgjdbc/src/main/java/org/postgresql/util/ReaderInputStream.java

```diff
@@ -142,7 +142,7 @@ public int read(byte b[], int off, int len) throws IOException {
         totalRead += remaining;
         off += remaining;
         len -= remaining;
-        if (len == 0) {
+        if (len >= 0) {
```

Comment:
This condition would effectively exit from the loop at the first iteration, so it will hurt performance.